internal/scheduler/scheduler.go

```diff
@@ -356,6 +356,54 @@ func (s *Scheduler) cycle(ctx *armadacontext.Context, updateAll bool, leaderToke
 	return overallSchedulerResult, nil
 }
 
+// syncInitialState updates jobs in jobDb to match state in postgres and returns all in-progress jobs.
+func (s *Scheduler) syncInitialState(ctx *armadacontext.Context) ([]*jobdb.Job, []jobdb.JobStateTransitions, error) {
+	txn := s.jobDb.WriteTxn()
+	defer txn.Abort()
+
+	// Load initial jobs and jobs runs from the jobRepo.
+	initialJobs, initialRuns, err := s.jobRepository.FetchInitialJobs(ctx)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Reconcile any differences between the updated jobs and runs.
+	jsts, err := s.jobDb.ReconcileDifferences(txn, initialJobs, initialRuns)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Upsert updated jobs (including associated runs).
+	jobDbJobs := make([]*jobdb.Job, 0, len(jsts))
+	for _, jst := range jsts {
+		if jst.Job != nil {
+			// We receive nil jobs from jobDb.ReconcileDifferences if a run is updated after the associated job is deleted.
+			// These nil job must be sorted out.
+			jobDbJobs = append(jobDbJobs, jst.Job)
+		}
+	}
+	if err := txn.Upsert(jobDbJobs); err != nil {
+		return nil, nil, err
+	}
+
+	latestJobSerial, latestJobRunSerial, err := s.jobRepository.FetchLatestSerials(ctx)
```

Comment:
I don't think this is quite right.  We fetch this in a separate database transaction from the initial load so we'll miss any updates that have occurred between `jobRepository.FetchInitialJobs` and here.

I think we should be able to derive `latestJobSerial` and `latestJobRunSerial` as the max serials of `initialJobs` and `initialRuns` respectively.

Reply:
Ah I see that now, you're correct I'll fix this up.